sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3937,6 +3937,25 @@ object SQLConf {
     .intConf
     .createWithDefault(20)
 
+  val ARTIFACTS_SESSION_ISOLATION_ENABLED =
+    buildConf("spark.sql.artifact.isolation.enabled")
+      .internal()
+      .doc("When enabled, artifacts (such as JARs, files, archives) added to one SparkSession " +
+        "are isolated from other sessions within the same Spark instance.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
What is the behaviour when some sessions have this disabled, and some have this enabled? Will all the sessions which disabled this flag share resources amongst each other?

Reply:
Yes, they'll be sharing artifacts under the `default` namespace.